lib/writeGltf.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var dataUri = require('datauri');
+var async = require('async');
+
+module.exports = writeGltf;
+
+function writeGltf(gltf, outputPath, isEmbedded, callback) {
+    var shaders = gltf.shaders;
+    
+    //Write out the source objects, followed by the entire gltf object
+    if (defined(shaders)) {
+        var shaderIds = Object.keys(shaders);
+        async.each(shaderIds, function(shaderId, asyncCallback) {
+            var shader = shaders[shaderId];
+            if (defined(shader.extras) && defined(shader.extras.source)) {
+                var source = shader.extras.source;
+                delete shader.extras.source;
+
+                var filePath = path.dirname(outputPath);
+                writeSource(filePath, shaderId, shader, source, isEmbedded, asyncCallback);
+            }
+        }, function(err) {
+            if (err) {
+                if (callback) {
+                        process.nextTick(function() {
+                        callback(err);
+                    });
+                }
+                else{
+                    throw err;
+                }
+            }
+            else if (callback) {

```

Comment:
We shouldn't call `callback` until we're inside the callback passed to `writeFile`, correct?